src/components/date-picker/dropdown.tsx

```diff
@@ -0,0 +1,131 @@
+import * as React from 'react';
+import {getMonthNames, getMonthFromOffset, getDayNames, getDaysInMonth, getNumOfPreviousDays, getNumOfFollowingDays} from '../../common/date-helpers';
+import {observable, action, computed} from 'mobx';
+import {observer} from 'mobx-react';
+import {Day} from './day';
+const styles = require('./date-picker.st.css').default;
+
+export interface DropdownProps {
+    value: Date;
+    onChange(date: Date): void;
+}
+
+const monthNames = getMonthNames();
+
+@observer
+export class Dropdown extends React.Component<DropdownProps, {}>{
+    @observable date: Date = this.props.value;
+
+    @action
+    setDate (date: Date) {
+        this.date = date;
+    }
+
+    @action
+    setDay = (day: number) => {
+        const date = new Date(this.date.getFullYear(), this.date.getMonth(), day);
+        this.props.onChange(date);
+    };
+
+    @computed
+    get monthName (): string {
+        return monthNames[this.date.getMonth()];
+    }
+
+    @computed
+    get year (): number {
+        return this.date.getFullYear();
+    }
+
+    @computed
+    get days (): Array<JSX.Element> {
+        const dayArray: Array<number> = [];
+        const daysInMonth = getDaysInMonth(this.date);
+
+        for (let i = 1; i <= daysInMonth; i++) {
+            dayArray.push(i);
+        }
+
+        return dayArray.map((day: number) => {
+            return (
+                <Day day={day}
+                     onSelect={this.setDay}
+                     dataAutomationId={'DAY_' + day}
+                     key={'DAY_' + day} />
+            );
+        });
+    }
```

Comment:
I suggest another approach: you can push the JSX elements directly into the dayArray and return it at the end.